lariat_agents/agent/s3_trigger/s3_trigger_agent.py

```diff
@@ -0,0 +1,97 @@
+from lariat_agents.base.base_agent import BaseAgent
+from boto3_type_annotations import s3
+import re
+import boto3
+import json
+from genson import SchemaBuilder
+from lariat_python_common.json.utils import get_clean_schema
+
+
+class S3TriggerAgent(BaseAgent):
+    def __init__(
+        self,
+        agent_type: str,
+        cloud: str,
+        s3_handler: s3.Client,
+        api_key: str = None,
+        application_key: str = None,
+    ):
+        self.s3_handler = s3_handler
+        super().__init__(
+            agent_type=agent_type,
+            cloud=cloud,
+            api_key=api_key,
+            application_key=application_key,
+            query_builder=None,
+        )
+
+    def schema_retrieval(self, event_dict):
```

Comment:
Assume that the config has a `file_type` param and for key json has a sub field called lines that is marked as `true` for jsonl case and `false` for non-lines case.